src/pages/contract/[cid].tsx

```diff
@@ -0,0 +1,230 @@
+
+function ContractSegmentOne({
+  creator,
+  balance,
+  setIsQrCodeClicked,
+}: {
+  creator: string;
+  balance: { value: number; symbol: string };
+  setIsQrCodeClicked: Dispatch<SetStateAction<boolean>>;
+}): JSX.Element {
+  const [isContractAddressCopied, setIsContractAddressCopied] = useState(false);
+  const router = useRouter();
+  const cid = router.query.cid?.toString()!;
+  return (
+    <div className="flex flex-col lg:gap-y-[37px] gap-y-[33px]">
+      <div>
+        {isContractAddressCopied ? (
+          <div className="flex flex-row gap-x-2.5 items-center">
+            <LinkText
+              testId="contract-address-copied"
+              label={fixedTitle.copied}
+              href={`/contract/${cid}`}
+              customStyle="tracking-[0.01em]"
+            />
+            <GreenTickIcon data-testid="contract-copied-green-tick-icon" />
+            <MdOutlineQrCode
+              role="button"
+              onClick={() => onQrCodeClick(setIsQrCodeClicked)}
+              className="text-white-50"
+            />
+          </div>
+        ) : (
+          <div className="flex flex-row gap-x-2.5 items-center">
+            <LinkText
+              testId="contract-address"
+              label={truncateTextFromMiddle(cid, 11)}
+              href={`/contract/${cid}`}
+              customStyle="tracking-[0.01em]"
+            />
+            <FiCopy
+              role="button"
+              data-testid="contract-address-copy-icon"
+              onClick={() =>
+                onCopyAddressIconClick(setIsContractAddressCopied, cid)
+              }
+              className="text-white-50"
+            />
+            <MdOutlineQrCode
+              data-testid="contract-address-qr-icon"
+              role="button"
+              onClick={() => onQrCodeClick(setIsQrCodeClicked)}
+              className="text-white-50"
+            />
+          </div>
+        )}
+      </div>
+      <div className="flex lg:flex-row md:flex-row flex-col gap-x-5 gap-y-5">
+        <div className="flex flex-col gap-y-1 w-[288px]">
+          <VerifiedContractSubtitle title={fixedTitle.creator} />
+          <LinkText
+            href={`/address/${creator}`}
+            label={truncateTextFromMiddle(creator, 11)}
+          />
+        </div>
+        <div className="flex flex-col gap-y-1">
+          <VerifiedContractSubtitle title={fixedTitle.balance} />
+          <NumericFormat
+            className="text-white-50 tracking-[0.01em]"
+            thousandSeparator
+            value={balance.value}
+            decimalScale={8}
+            suffix={` ${balance.symbol}`}
+            data-testid="contract-wallet-balance-value"
+          />
+        </div>
+      </div>
+    </div>
+  );
+}
+
+function ContractSegmentTwo({ data }) {
+  const [isTransactionClicked, setIsTransactionClicked] = useState(false);
+  const [isCodeClicked, setIsCodeClicked] = useState(true);
+  const [isTokenClicked, setIsTokenClicked] = useState(false);
```

Comment:
Imagine today if we have 5 tabs, or more in the future, creating more boolean flags is not a scalable solutions.
Try to combine all the state as one, something like `activeTab`, create an enum for each new tab